lti_consumer/plugin/views.py

```diff
@@ -44,14 +46,21 @@ def public_keyset_endpoint(request, usage_id=None):
     and run the proper handler.
     """
     try:
-        usage_key = UsageKey.from_string(usage_id)
-
-        return run_xblock_handler_noauth(
-            request=request,
-            course_id=str(usage_key.course_key),
-            usage_id=str(usage_key),
-            handler='public_keyset_endpoint'
+        lti_config = LtiConfiguration.objects.get(
+            location=UsageKey.from_string(usage_id)
         )
+
+        if lti_config.version != lti_config.LTI_1P3:
+            raise Exception(
```

Comment:
@giovannicimolin  Shouldn't we use ``LtiError`` type here? https://github.com/edx/xblock-lti-consumer/blob/master/lti_consumer/exceptions.py#L6

Reply:
I will second this